scripts/upgrade.js

```diff
@@ -0,0 +1,40 @@
+const assert = require("assert");
+const path = require('path');
+const fs = require('fs');
+
+const address = "0xDc64a140Aa3E981100a9becA4E685f962f0cF6C9";
+
+(async () => {
+    const stringParserLibraryFactory = await ethers.getContractFactory("contracts/StringParserLibrary.sol:StringParserLibrary");
+    const stringParserLibrary = await stringParserLibraryFactory.deploy();
+
+    const domainParserLibraryFactory = await ethers.getContractFactory("contracts/DomainParserLibrary.sol:DomainParserLibrary", {
+        libraries: {
+            StringParserLibrary: stringParserLibrary
+        }
+    });
+    const domainParserLibrary = await domainParserLibraryFactory.deploy();
+    const artifactsPath = path.resolve(__dirname, '../artifacts/contracts/DomainRegistry.sol/DomainRegistry.json');
+    const contractABI = JSON.parse(fs.readFileSync(artifactsPath, 'utf8'));
+
+    const oldContract = new ethers.Contract(address, contractABI.abi);
+
+    const domainRegistryV2 = await ethers.getContractFactory("DomainRegistry", {
+        libraries: {
+            DomainParserLibrary: domainParserLibrary,
+        }
+    });
+    const domainsContractV2 = await upgrades.upgradeProxy(oldContract, domainRegistryV2, {
+        call: {
+            fn: "reinitialize"
+        },
+        unsafeAllowLinkedLibraries: true,
+    });
+
+    console.log("Upgraded DomainsContract contains new functionality");
+    const value = await domainsContractV2.REWARD_PERCENT_OWNER();
+    assert(BigInt(value) === BigInt(10));
+
+    console.log("address:", address);
+    console.log("upgradedToContractV2 address:", await domainsContractV2.getAddress());
```

Comment:
No this is not enough, to check upgrade safety. You didn't populate contract with some data before the upgrade, and checked the consistency after the upgrade. For example register parent domain before upgrade, register subdomain after upgrade and check if parent got some ether. I described the process in the "info" Slack chat.
   -3 points ( for not completely understanding how the upgrade works)